django/template/base.py

```diff
@@ -920,12 +949,25 @@ def render_annotated(self, context):
         where it occurred in the template. For internal usage this method is
         preferred over using the render method directly.
         """
-        try:
+        with annotate_exception(context, self.token):
             return self.render(context)
-        except Exception as e:
-            if context.template.engine.debug and not hasattr(e, 'template_debug'):
-                e.template_debug = context.template.get_exception_info(e, self.token)
-            raise
+
+    def stream(self, context):
+        """
+        Return a string generator rendering the node.
+        """
+        yield self.render(context)
+
+    def stream_annotated(self, context):
+        """
+        Render the node. If debug is True and an exception occurs during
+        rendering, the exception is annotated with contextual line information
+        where it occurred in the template. For internal usage this method is
+        preferred over using the render method directly.
+        """
```

Comment:
This docstring still uses ``render`` rather than ``stream``.

I think it would help to move part of these docstrings to ``annotate_exception``. Example:

```diff
 @contextmanager
 def annotate_exception(context, token):
+    """
+    Checks for errors during template rendering. If debug is True, contextual
+    line information is added to the exception before it is reraised.
+    """
     try:
         yield
     except Exception as e:
@@ -944,10 +942,9 @@ class Node(object):
 
     def render_annotated(self, context):
         """
-        Render the node. If debug is True and an exception occurs during
-        rendering, the exception is annotated with contextual line information
-        where it occurred in the template. For internal usage this method is
-        preferred over using the render method directly.
+        Render the node with debug information if an error occurs. For
+        internal usage this method is preferred over using the render method
+        directly.
         """
         with annotate_exception(context, self.token):
             return self.render(context)
@@ -960,10 +957,9 @@ class Node(object):
 
     def stream_annotated(self, context):
         """
-        Render the node. If debug is True and an exception occurs during
-        rendering, the exception is annotated with contextual line information
-        where it occurred in the template. For internal usage this method is
-        preferred over using the render method directly.
+        Stream the node with debug information if an error occurs. For
+        internal usage this method is preferred over using the render method
+        directly.
         """
         with annotate_exception(context, self.token):
             for chunk in self.stream(context):
```